### PR TITLE
Add and test support for Convertable scalar and parameter values

### PR DESCRIPTION
**Pull request by tjibbechris**

### CODE REVIEW

File: src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs [hunk 1]
```diff
@@ -26,7 +26,17 @@ public static T ReadScalar<T>(this ParameterizedSql sql, [NotNull] SqlCommandCre
         {
             using (var cmd = sql.CreateSqlCommand(commandCreationContext)) {
                 try {
-                    return DBNullRemover.Cast<T>(cmd.Command.ExecuteScalar());
+                    var converter = MetaObjectPropertyConverter.GetOrNull(typeof(T).GetNonNullableType());
+                    if (converter == null ) {
+                        return DBNullRemover.Cast<T>(cmd.Command.ExecuteScalar());
+                    }
+                    {
```
Comment (EamonNerbonne) [inline src/ProgressOnderwijsUtils/Data/ParameterizedSqlObjectMapper.cs]: syntax weirdness.

File: tools/ProgressOnderwijsUtilsBenchmarks/MicroOrmBench/Program.cs [hunk 2]
```diff
@@ -16,11 +16,11 @@ static void RunCurrentBenchmarks([NotNull] Benchmarker benchmarker)
         {
             ParameterizedSqlExecutor.ConstructWithoutExecuting(benchmarker);
             ParameterizedSqlExecutor.RunQuery(benchmarker);
+            ParameterizedSqlExecutor.RunWideQuery(benchmarker);
             //ParameterizedSqlExecutor.RunTvpQuery(benchmarker);
             //DapperExecutor.RunQuery(benchmarker);
-            //ParameterizedSqlExecutor.RunWideQuery(benchmarker);
-            //HandrolledAdoNetExecutor.RunQuery(benchmarker);
-            //HandrolledAdoNetExecutor.RunWideQuery(benchmarker);
+            HandrolledAdoNetExecutor.RunQuery(benchmarker);
```
Comment (tjibbechris) [inline tools/ProgressOnderwijsUtilsBenchmarks/MicroOrmBench/Program.cs]: Sould these be left as is?

Comment (EamonNerbonne) [reply tools/ProgressOnderwijsUtilsBenchmarks/MicroOrmBench/Program.cs]: Mij maakt het niet uit.  Typisch doen we dev in bursts, dus MRU is geen gekke caching strategy, en dat is het laten staan een beetje ;-).

File: test/ProgressOnderwijsUtils.Tests/TableValuedParameterTest.cs [hunk 3]
```diff
@@ -18,6 +20,45 @@ public sealed class TableValuedParameterTest : TransactedLocalConnection
...
+            var value = SQL($@"select {new CustomBlaStruct("aap")}").ReadScalar<CustomBlaStruct?>(Context);
+            PAssert.That(() => value.Value.CustomBlaString == "aap");
+        }
+
+        [Fact]
+        public void ConvertibleNullablePropertyWithoutValue()
+        {
+            var value = SQL($@"select {default(CustomBlaStruct?)}").ReadScalar<CustomBlaStruct?>(Context);
+            PAssert.That(() => value == null);
+        }
```
Comment (EamonNerbonne) [inline test/ProgressOnderwijsUtils.Tests/TableValuedParameterTest.cs]: dus ik denk dat we hier ook extra tests willen voor null-containing struct, die volgens mij moet crashen.

Comment (EamonNerbonne) [reply test/ProgressOnderwijsUtils.Tests/TableValuedParameterTest.cs]: ik heb een test toegevoegd.
